sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeed.cs

```diff
@@ -0,0 +1,389 @@
+        {
+            _containerClient = blobServiceClient.GetBlobContainerClient(Constants.ChangeFeed.ChangeFeedContainerName);
+            _years = new Queue<string>();
+            _segments = new Queue<string>();
+            _isInitalized = false;
+            _startTime = startTime.RoundDownToNearestHour();
+            _endTime = endTime.RoundUpToNearestHour();
+        }
+
+        public ChangeFeed(
+            BlobServiceClient blobServiceClient,
+            string continutation)
+        {
+            _containerClient = blobServiceClient.GetBlobContainerClient(Constants.ChangeFeed.ChangeFeedContainerName);
+            ChangeFeedCursor cursor = JsonSerializer.Deserialize<ChangeFeedCursor>(continutation);
+            ValidateCursor(_containerClient, cursor);
+            _years = new Queue<string>();
+            _segments = new Queue<string>();
+            _isInitalized = false;
+            _startTime = cursor.CurrentSegmentCursor.SegmentTime;
+            _endTime = cursor.EndTime;
+            _currentSegmentCursor = cursor.CurrentSegmentCursor;
+        }
+
+        /// <summary>
+        /// Internal constructor for unit tests.
+        /// </summary>
+        /// <param name="containerClient"></param>
+        internal ChangeFeed(
+            BlobContainerClient containerClient)
+        {
+            _containerClient = containerClient;
+        }
+
+        private async Task Initalize(bool async)
+        {
+            // Check if Change Feed has been abled for this account.
+            bool changeFeedContainerExists;
+
+            if (async)
+            {
+                changeFeedContainerExists = await _containerClient.ExistsAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                changeFeedContainerExists = _containerClient.Exists();
+            }
+
+            if (!changeFeedContainerExists)
+            {
+                //TODO improve this error message
+                throw new ArgumentException("Change Feed hasn't been enabled on this account, or is current being enabled.");
+            }
+
+            // Get last consumable
+            BlobClient blobClient = _containerClient.GetBlobClient(Constants.ChangeFeed.MetaSegmentsPath);
+            BlobDownloadInfo blobDownloadInfo;
+            if (async)
+            {
+                blobDownloadInfo = await blobClient.DownloadAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                blobDownloadInfo = blobClient.Download();
+            }
+
+            JsonDocument jsonMetaSegment;
+            if (async)
+            {
+                jsonMetaSegment = await JsonDocument.ParseAsync(blobDownloadInfo.Content).ConfigureAwait(false);
+            }
+            else
+            {
+                jsonMetaSegment = JsonDocument.Parse(blobDownloadInfo.Content);
+            }
+
+            //TODO what happens when _lastConsumable advances an hour?
```

Comment:
Maybe we should just default end time to "now" when change feed is requested ?

Reply:
I think we just default to the last hour present in the Change Feed.